File: Go/Algorithms/2220.go
package main

func minBitFlips(start int, goal int) int {
	count, b, s := 0, start, goal
	if goal > start {
		b, s = s, b
	}

	for s > 0 {
		if s&1 != b&1 {
			count++
		}
		s >>= 1
		b >>= 1
	}
	for b > 0 {
		if b&1 == 1 {
			count++
		}
		b >>= 1
	}
	return count
}

File: Rust/Algorithms/2220.rs
impl Solution {
    pub fn min_bit_flips(start: i32, goal: i32) -> i32 {
        let mut count = 0;
        let (mut b, mut s) = if start > goal { (start, goal) } else { (goal, start) };
    
        while s > 0 {
            if b & 1 != s & 1 {
                count += 1;
            }
            b >>= 1;
            s >>= 1;
        }
        
        while b > 0 {
            if b & 1 == 1 {
                count += 1;
            }
            b >>= 1;
        }
        count
    }
}